spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -314,6 +316,11 @@ public void toHeaders(Headers source, final Map<String, Object> headers) {
 			else if (headerName.equals(KafkaHeaders.LISTENER_INFO) && matchesForInbound(headerName)) {
 				headers.put(headerName, new String(header.value(), getCharset()));
 			}
+			else if ((headerName.equals(SerializationUtils.KEY_DESERIALIZER_EXCEPTION_HEADER) ||
+					headerName.equals(SerializationUtils.VALUE_DESERIALIZER_EXCEPTION_HEADER))
+					&& matchesForInbound(headerName)) {
```

Comment:
I believe this header has to be mapped unconditionally. It is not some what end-user provides.